src/python_testing/tc_mode_base.py

```diff
@@ -0,0 +1,139 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+from mobly import asserts
+from chip.clusters.Types import NullValue
+
+logger = logging.getLogger(__name__)
+
+# Maximum value for ModeTags according to specs is 16bits.
+MAX_MODE_TAG = 0xFFFF
+# According to specs, the specific MfgTags should be defined in the range 0x8000 - 0xBFFF
+START_MFGTAGS_RANGE = 0x8000
+END_MFGTAGS_RANGE = 0xBFFF
+
+
+class ClusterModeCheck:
+    """ Class that holds the common Mode checks between TCs
+
+    Several TCs have similar checks in place for functionality that is common among them.
+    This class holds most of this common functionality to avoid duplicating code with the same validations.
+
+    Link to spec:
+    https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/cluster/modebase_common.adoc
+
+
+    Attributes:
+        requested_cluster: A reference to the cluster to be tested, it should be a derived from the Mode Base cluster.
```

Comment:
nit: consider `modebase_derived_cluster` (since the documentation says this `should be a derived from the Mode Base cluster`)?

Reply:
That makes sense, changed the name of the parameter.